app/src/test/java/org/kinecosystem/kinit/viewmodel/RestoreViewModelTest.kt

```diff
@@ -0,0 +1,123 @@
+package org.kinecosystem.kinit.viewmodel
+
+import com.google.common.truth.Truth.assertThat
+import kin.core.KinAccount
+import org.junit.Before
+import org.junit.Test
+import org.kinecosystem.kinit.KinitApplication
+import org.kinecosystem.kinit.blockchain.Wallet
+import org.kinecosystem.kinit.daggerCore.TestCoreComponentProvider
+import org.kinecosystem.kinit.repository.UserRepository
+import org.kinecosystem.kinit.server.ERROR_APP_SERVER_FAILED_RESPONSE
+import org.kinecosystem.kinit.server.OnboardingService
+import org.kinecosystem.kinit.server.OperationCompletionCallback
+import org.kinecosystem.kinit.viewmodel.restore.RestoreWalletActivityMessages
+import org.kinecosystem.kinit.viewmodel.restore.RestoreWalletViewModel
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito.*
+import org.mockito.MockitoAnnotations
+import javax.inject.Inject
+
+
+private const val QR_SAMPLE_CODE = "abcdefghijklmnopqrstuvwxyzzyxwvutsrqponmlkjihgfedcba"
+private const val QR_SAMPLE_CODE_ANSWER_ONE = "1234"
+private const val QR_SAMPLE_CODE_ANSWER_TWO = "1234"
+
+class RestoreViewModelTest {
+    @Inject
+    lateinit var userRepository: UserRepository
+    @Inject
+    lateinit var onboardingService: OnboardingService
+    @Inject
+    lateinit var walletService: Wallet
+
+    private lateinit var restoreWalletViewModel: RestoreWalletViewModel
+    private lateinit var coreComponentProvider: TestCoreComponentProvider
+
+    @Before
+    fun setup() {
+        MockitoAnnotations.initMocks(this)
+        coreComponentProvider = TestCoreComponentProvider()
+        KinitApplication.coreComponent = coreComponentProvider.coreComponent
+        coreComponentProvider.coreComponent.inject(this)
+
+        `when`(userRepository.restoreHints).thenReturn(listOf("1", "2"))
+        restoreWalletViewModel = RestoreWalletViewModel()
+    }
+
+    @Test
+    fun nextEnabled_False_When_setAnswer_shortInput() {
+        restoreWalletViewModel.setAnswer(0, "")
+        assertThat(restoreWalletViewModel.nextEnabled.get()).isFalse()
+        restoreWalletViewModel.setAnswer(0, "1234")
+        assertThat(restoreWalletViewModel.nextEnabled.get()).isFalse()
+        restoreWalletViewModel.setAnswer(0, "")
+        restoreWalletViewModel.setAnswer(1, "1234")
+        assertThat(restoreWalletViewModel.nextEnabled.get()).isFalse()
+    }
+
+    @Test
+    fun nextEnabled_True_When_setAnswer_longInput() {
+        restoreWalletViewModel.setAnswer(0, "1234")
+        restoreWalletViewModel.setAnswer(1, "1234")
+        assertThat(restoreWalletViewModel.nextEnabled.get()).isTrue()
```

Comment:
I would also add a test to check that we do not count leading or tailing spaces in the input (i.e. that the input is indeed trimmed)